_posts/2016-10-07-our-jekyll-workflow.md

```diff
@@ -0,0 +1,903 @@
+
+The default task deletes the site by running the `clean` tasks then runs all the
+`build` tasks, then the `jekyll build` command to build the site using our
+processed assets.
+
+A little gulp tip: tasks passed within square brackets are run concurrently (one
+of the major advantages of using gulp over Grunt, which can only run one task at
+a time), while items in a comma-separated list are run sequentially. We'll need
+this to run the build tasks in the correct order since we need to create our
+`assets` directory before running `build:jekyll`, which will copy `assets` to
+`_site/assets`.
+
+```js
+// Runs jekyll build command.
+gulp.task('build:jekyll', function() {
+    var shellCommand = 'bundle exec jekyll build --config _config.yml';
+
+    return gulp.src('')
+        .pipe(run(shellCommand))
+        .on('error', gutil.log);
+});
+
+// Deletes the entire _site directory.
+gulp.task('clean:jekyll', function(callback) {
+    del(['_site']);
+    callback();
+});
+
+// Main clean task.
+// Deletes _site directory and processed assets.
+gulp.task('clean', ['clean:jekyll',
+    'clean:fonts',
+    'clean:images',
+    'clean:scripts',
+    'clean:styles']);
+
+// Builds site anew.
+gulp.task('build', function(callback) {
+    runSequence('clean',
+        ['build:scripts', 'build:images', 'build:styles', 'build:fonts'],
+        'build:jekyll',
+        callback);
+});
+
+// Default Task: builds site.
+gulp.task('default', ['build']);
+```
+
+Now, `gulp` will run the default task `build` and will build the `_site`
+directory. This is what we want in production, but let's go back to our three
+separate Jekyll configuration files. We want to create gulp tasks using
+`config.test.yml` for running our test script and `config.dev.yml` for local
+development.
+
+```js
+// Runs jekyll build command using test config.
+gulp.task('build:jekyll:test', function() {
+    var shellCommand = 'bundle exec jekyll build --config _config.yml,_config.test.yml';
+
+    return gulp.src('')
+        .pipe(run(shellCommand))
+        .on('error', gutil.log);
+});
+
+// Runs jekyll build command using local config.
+gulp.task('build:jekyll:local', function() {
+    var shellCommand = 'bundle exec jekyll build --config _config.yml,_config.test.yml,_config.dev.yml';
+
+    return gulp.src('')
+        .pipe(run(shellCommand))
+        .on('error', gutil.log);
+});
+```
+
+The last thing we needed is a gulp command to serve the site and watch for file
+changes. To accomplish this we're using `gulp.watch()`, which tells gulp to run
+a specified task if the targeted files are changed. We also created a couple of
+special tasks to tell the browser to reload the served site on changes to Jekyll
+files (templates, config, etc.) or Javascript files. We already included
+`.pipe(browserSync.stream())` in our `build:styles:main` and `build:images` tasks, so
+when these files are updated the changes will be injected into our local site so
+we won't even need a refresh.
```

Comment:
Maybe it'd be better like this : "..so when these files are updated the changes will be injected into our local site **and** we won’t even need a refresh." - your call.